src/main/java/core/basesyntax/strategy/impl/PurchaseOperationImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperationImpl implements OperationHandler {
+    @Override
+    public void applyOperation(FruitTransaction transaction) {
+        if (transaction.getQuantity() <= MIN_QUANTITY) {
+            throw new RuntimeException("Purchase must be 1 or more items. But was: "
+                    + transaction.getQuantity());
+        }
+        if ((getCurrentBalance(transaction) - transaction.getQuantity()) < 0) {
+            throw new RuntimeException("There is no required quantity in stock.Exist: "
+                    + getCurrentBalance(transaction) + "Required: "
+                    + transaction.getQuantity());
```

Comment:
The message 'There is no required quantity in stock.Exist:' has a typo and formatting issue. There should be a space after 'stock.' and before 'Exist:'. Also, the word 'Required' should be lowercased and followed by a colon and a space for consistency in the message format.